### PR TITLE
Fix #7402 Upgrade jQuery and Moment.js

### LINKED ISSUE #7402: upgrade Jquery and moment

### PR DESCRIPTION
**Pull request by dennishendriksen**
#### Checklist
- [x] Functionality works & meets specifications
- [x] Code reviewed
- [x] Code unit/integration/system tested
- [x] User documentation updated
- [x] (If you have changed REST API interface) view-swagger.ftl updated
- [x] Test plan template updated
- [x] Clean commits (squash)
- (n.a.) Added Feature/Fix to release notes

### CODE REVIEW

**Comment by fdlk** [comment]
test this please

File: molgenis-semantic-mapper/src/main/resources/js/attribute-mapping.js [hunk 1]
```diff
@@ -681,6 +681,7 @@
                     }
                 } else {
                     if (val !== 'use-default-option') {
+                        removeAttr
```
Comment (fdlk) [inline molgenis-semantic-mapper/src/main/resources/js/attribute-mapping.js]: hum?

**Comment by fdlk** [comment]
The [jquery migrate readme](https://github.com/jquery/jquery-migrate) says we first need to upgrade to jquery 2.2 before using the jquery migrate 1.x plugin.

**Comment by dennishendriksen** [comment]
> The jquery migrate readme says we first need to upgrade to jquery 2.2 using the jquery migrate 1.x plugin, before switching to this plugin and upgrading to jquery 3.

Yup, I'm aware of that. See https://blog.jquery.com/2016/01/08/jquery-2-2-and-1-12-released/ section 'Full changelogs' for all 2.1 > 2.2 changes.

**Comment by fdlk** [comment]
So (how?) are we going test both version changes in sequence using both migration plugins?

**Comment by dennishendriksen** [comment]
> So (how?) are we going test both version changes in sequence using both migration plugins?

I've tested the 2.1 to 2.2 migration locally using the migration plugin focusing on import, exploring, filtering, modals, alters, mapping service and checked the 2.1 > 2.2 changes that I mentioned in my previous message to look for relevant breaking changes. None discovered.

The migration to 3.0 resulted in some selector syntax issues that are fixed in this PR.

**Comment by fdlk** [comment]
test this please

**Comment by dennishendriksen** [comment]
test this please

**Comment by dennishendriksen** [comment]
test this please

File: molgenis-core-ui/package.json [hunk 2]
```diff
@@ -24,17 +24,18 @@
     "bootstrap": "3.3.6",
     "bootstrap-fileinput": "4.2.8",
     "brace": "0.5.1",
-    "eonasdan-bootstrap-datetimepicker": "4.15.35",
+    "eonasdan-bootstrap-datetimepicker": "4.17.47",
     "handlebars": "4.0.5",
     "javascript-natural-sort": "0.7.1",
-    "jquery": "2.1.1",
+    "jquery": "3.3.1",
+    "jquery-migrate": "3.0.1",
```
Comment (fdlk) [inline molgenis-core-ui/package.json]: Does not work with webpack, see https://github.com/jquery/jquery-migrate/issues/273

**Comment by fdlk** [comment]
jquery ui results in some warnings by jquery migrate, but they sound harmless
See https://stackoverflow.com/a/46035524/1973271

File: molgenis-core-ui/package.json [hunk 3]
```diff
@@ -24,17 +24,18 @@
...
     "jquery-deparam": "0.5.2",
     "jquery-form": "3.50.0",
     "jquery-validation": "1.14.0",
     "jquery.cookie": "^1.4.1",
     "jquery.fancytree": "2.14.0",
     "mdurl": "^1.0.1",
-    "moment": "2.11.2",
+    "moment": "^2.20.1",
```
Comment (fdlk) [inline molgenis-core-ui/package.json]: Still have a static moment v2.9.0 in src/main/resources/js of core-ui

**Comment by fdlk** [comment]
Do we want the migrate plugin included in production, and if so do we want it to log?

File: molgenis-core-ui/package.json (hunk 3 above)
Comment (fdlk) [reply molgenis-core-ui/package.json]: I've removed it.

**Pull request merged**